### PR TITLE
Add extractors for CaracolTvPlay and Canal1

### LINKED ISSUE #5826: Add play.Caracoltv.com Support
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Colombia

### Example URLs

- Single chapter: https://play.caracoltv.com/play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ%3D%3D&season=0&episode=0
- Entire series: https://play.caracoltv.com/videoDetails/OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==?season=0&episode=0
- Livestream: https://play.caracoltv.com/play?videoId=MzoxX3BwbjRmNjB1&mode=channel

### Provide a description that is worded well enough to be understood

1. you can only access the content with an account (making an account is free, so there is no problem in this section).
2. I am almost sure it is blocked by geography but this may be false.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://play.caracoltv.com/play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ%3D%3D']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] ytdl-patched/yt-dlp version 2022.12.18.334 [391c11e] (win_exe)
[debug] ** This build is unofficial daily builds, provided for ease of use.
[debug] ** Please do not ask for any support.
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2022-08-13-git-c469c3c3b1-full_build-www.gyan.dev (setts), ffprobe 2022-08-13-git-c469c3c3b1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1735 extractors
[debug] Fetching release info: https://api.github.com/repos/ytdl-patched/yt-dlp/releases/latest
Latest version: 2022.12.18.334, Current version: 2022.12.18.334
yt-dlp is up to date (2022.12.18.334)
[generic] Extracting URL: https://play.caracoltv.com/play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ%3D%3D
[generic] play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://play.caracoltv.com/play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ%3D%3D
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1495, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1571, in __extract_info
  File "yt_dlp\extractor\common.py", line 680, in extract
  File "yt_dlp\extractor\generic.py", line 2453, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://play.caracoltv.com/play?seriesId=OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ%3D%3D
```

### PR DESCRIPTION
**Pull request by elyse0**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This PR adds extractors for CaracolTvPlay and Canal1 which are two sites owned by public TV channels in Colombia

The CaracolTvPlay app uses some default credentials to display the catalog and this is enough to extract these videos.  It should also be possible to pass your own username and password, but I haven't tested it myself

Canal1 extractor just extracts the embedUrl field

Some content in these extractors is geo-blocked, but hopefully it will work without problems inside the region

Fixes #5826
Related #5929 (The issue is now about supporting JSON LD embedUrl field in the generic extractor)

#### CaracolTvPlay series

```bash
[debug] Command-line config: ['https://play.caracoltv.com/videoDetails/OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==', '--verbose', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 9dd2004b4
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-42-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg N-108931-g4dda3b1653-20221104 (setts), ffprobe N-108931-g4dda3b1653-20221104, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: ElyseNebulaChannelIE, ElyseNebulaIE, ElyseNebulaSubscriptionsIE, OnlineRadioBoxIE
[debug] Plugin directories: ['/home/amish/.yt-dlp/plugins/elyse0/yt_dlp_plugins']
[debug] Loaded 1827 extractors
[CaracolTvPlay] Extracting URL: https://play.caracoltv.com/videoDetails/OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==
[CaracolTvPlay] None: Downloading webpage
[CaracolTvPlay] Extracting JS config
[CaracolTvPlay] Retrieving bearer token
[CaracolTvPlay] Performing login
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading JSON metadata
[download] Downloading playlist: La teoría del promedio
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading JSON metadata
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] OTo4NGFmNjUwOWQ2ZmM0NTg2YWRiOWU0MGNhOWViOWJkYQ==: Downloading m3u8 information
[CaracolTvPlay] Playlist La teoría del promedio: Downloading 6 items of 6
[download] Downloading item 1 of 6
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for MjoxX3ByaXplOWJi:
ID   EXT RESOLUTION │   TBR PROTO │ VCODEC        VBR ACODEC
───────────────────────────────────────────────────────────────
129  mp4 256x144    │  129k m3u8  │ avc1.42000d  129k mp4a.40.5
275  mp4 426x240    │  275k m3u8  │ avc1.42000d  275k mp4a.40.5
413  mp4 640x360    │  413k m3u8  │ avc1.4d001e  413k mp4a.40.5
635  mp4 854x480    │  635k m3u8  │ avc1.4d001e  635k mp4a.40.5
1262 mp4 1280x720   │ 1262k m3u8  │ avc1.4d001f 1262k mp4a.40.5
3052 mp4 1920x1080  │ 3052k m3u8  │ avc1.640028 3052k mp4a.40.5
[download] Downloading item 2 of 6
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for MjoxX2xicWlra2c5:
ID   EXT RESOLUTION │   TBR PROTO │ VCODEC        VBR ACODEC
───────────────────────────────────────────────────────────────
140  mp4 256x144    │  140k m3u8  │ avc1.42000d  140k mp4a.40.5
298  mp4 426x240    │  298k m3u8  │ avc1.42000d  298k mp4a.40.5

```

#### Canal1 extractor
```bash
[debug] Command-line config: ['https://canal1.com.co/noticias/napa-i-una-cadena-de-produccion-de-arroz-que-se-quedo-en-veremos-y-abandonada-en-el-departamento-del-choco/', '--verbose', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 9dd2004b4
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-42-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg N-108931-g4dda3b1653-20221104 (setts), ffprobe N-108931-g4dda3b1653-20221104, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: ElyseNebulaChannelIE, ElyseNebulaIE, ElyseNebulaSubscriptionsIE, OnlineRadioBoxIE
[debug] Plugin directories: ['/home/amish/.yt-dlp/plugins/elyse0/yt_dlp_plugins']
[debug] Loaded 1827 extractors
[Canal1] Extracting URL: https://canal1.com.co/noticias/napa-i-una-cadena-de-produccion-de-arroz-que-se-quedo-en-veremos-y-abandonada-en-el-departamento-del-choco/
[Canal1] napa-i-una-cadena-de-produccion-de-arroz-que-se-quedo-en-veremos-y-abandonada-en-el-departamento-del-choco: Downloading webpage
[MediaStream] Extracting URL: https://mdstrm.com/embed/63b39f6b354977084b85ab54
[MediaStream] 63b39f6b354977084b85ab54: Downloading webpage
[MediaStream] 63b39f6b354977084b85ab54: Downloading m3u8 information
[MediaStream] 63b39f6b354977084b85ab54: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 63b39f6b354977084b85ab54:
ID    EXT RESOLUTION FPS │  TBR PROTO │ VCODEC       VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────
a1-x3 m4a audio only     │ 131k dash  │ audio only       mp4a.40.2  131k 48k DASH audio, m4a_dash
1     mp4 unknown        │      https │ unknown          unknown
520   mp4 640x360        │ 520k m3u8  │ avc1.4d001e 520k mp4a.40.5
v1-x3 mp4 640x360     30 │ 520k dash  │ avc1.4d401e 520k video only          DASH video, mp4_dash
```

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🇨🇴📺🐛

This pull request adds support for downloading videos from two Colombian websites, Canal 1 and Caracol TV Play, by creating new extractors `Canal1IE` and `CaracolTvPlayIE` in the `canal1.py` and `caracoltv.py` modules. It also improves the geo-restriction handling of the existing extractor `MediaStreamIE` in the `mediastream.py` module.

> _We're sailing the web for videos to grab_
> _From `Canal1IE` and `CaracolTvPlayIE`_
> _We'll use the `MediaStreamIE` to check the flag_
> _And download them all with `youtube-dl` (hey!)_

### Walkthrough
*  Add support for two Colombian websites, Canal 1 and Caracol TV Play, as new extractors ([link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R291), [link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R301), [link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-1e71048bc941e11325f2f5cb28b441c4d6fd075d7f7fcdc99a0e1ac8d0b42933R1-R39), [link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-49b2e2f7eb4a70412c3aefead62f1331319120cdc341f04d1999477d1700b1e7R1-R139))
   * Import the `Canal1IE` and `CaracolTvPlayIE` classes from their respective modules in `_extractors.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R291), [link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R301))
   * Define the methods and attributes for extracting videos from Canal 1 in `canal1.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-1e71048bc941e11325f2f5cb28b441c4d6fd075d7f7fcdc99a0e1ac8d0b42933R1-R39))
   * Define the methods and attributes for extracting videos from Caracol TV Play in `caracoltv.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-49b2e2f7eb4a70412c3aefead62f1331319120cdc341f04d1999477d1700b1e7R1-R139))
* Improve the detection of geo-restricted videos hosted on MediaStream by adding another possible error message in `mediastream.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7151/files?diff=unified&w=0#diff-b9028f71bb7b6a439ee2f3a7ce36c099a1fb3444c3a86fdaea97fd0882219d0cL109-R114))

</details>

### CODE REVIEW

File: yt_dlp/extractor/caracoltv.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        return base64.b64encode(f'{key}:{secret}'.encode()).decode()
+
+    def _login(self, email, password):
+        webpage = self._download_webpage('https://play.caracoltv.com/', None, fatal=False)
+        app_token = self._extract_app_token(webpage)
+
+        bearer_token = self._download_json(
+            'https://eu-gateway.inmobly.com/applications/oauth', None, data={}, note='Retrieving bearer token',
+            headers={'Authorization': f'Basic {app_token}'})['token']
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:71]: `data={}` will not work with networking overhaul
```suggestion
        bearer_token = self._download_json(
            'https://eu-gateway.inmobly.com/applications/oauth', None, data=b'', note='Retrieving bearer token',
            headers={'Authorization': f'Basic {app_token}'})['token']
```

File: yt_dlp/extractor/caracoltv.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            'mediation_config', None, transform_source=js_to_json, fatal=False)
+
+        key = traverse_obj(
+            mediation_config, ('live', 'key')) or '795cd9c089a1fc48094524a5eba85a3fca1331817c802f601735907c8bbb4f50'
+        secret = traverse_obj(
+            mediation_config, ('live', 'secret')) or '64dec00a6989ba83d087621465b5e5d38bdac22033b0613b659c442c78976fa0'
+
+        return base64.b64encode(f'{key}:{secret}'.encode()).decode()
+
+    def _login(self, email, password):
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:65]:
```suggestion
    def _perform_login(self, email, password):
```

File: yt_dlp/extractor/caracoltv.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                'login_data': {
+                    'enabled': True,
+                    'email': email,
+                    'password': password,
+                }
+            }).encode())['user_token']
+
+    def _perform_login(self, username, password):
+        self._login(username, password)
+
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:92]:
```suggestion
```

File: yt_dlp/extractor/caracoltv.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            headers={'Authorization': 'Bearer ' + self._USER_TOKEN})
+
+        for episode in api_response['items']:
+            yield self._extract_video(episode, series_id, season_id, season_number)
+
+    def _real_extract(self, url):
+        series_id = self._match_id(url)
+
+        if self._USER_TOKEN is None:
+            self._login('guest@inmobly.com', 'Test@gus1')
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:123]:
```suggestion
            self._perform_login('guest@inmobly.com', 'Test@gus1')
```

File: yt_dlp/extractor/caracoltv.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            'https://eu-gateway.inmobly.com/feed', series_id, query={'include_ids': series_id},
+            headers={'Authorization': 'Bearer ' + self._USER_TOKEN})['items'][0]
+
+        if not api_response.get('seasons'):
+            return self._extract_video(api_response)
+
+        return self.playlist_result(
+            itertools.chain.from_iterable(
+                self._extract_series_season(series_id, season['id'], season.get('order'))
+                for season in api_response['seasons']),
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:135]: can this be a `PagedList`?

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/caracoltv.py (hunk 5 above)
Comment (elyse0) [reply yt_dlp/extractor/caracoltv.py:133]: Mmm I don't think so, there is no page size nor page number

Comment (Grub4K) [reply yt_dlp/extractor/caracoltv.py:133]: How about using
```py
    def _extract_series_seasons(self, seasons, series_id):
        for season in seasons:
            api_response = self._download_json(
                'https://eu-gateway.inmobly.com/feed', series_id, query={'season_id': season['id']},
                headers={'Authorization': f'Bearer {self._USER_TOKEN}'})

            season_number = season.get('order')
            for episode in api_response['items']:
                yield self._extract_video(episode, series_id, season['id'], season_number)
```
Then we can simplify this here. I think thats easier to parse than the `itertools.chain`
```suggestion
            self._extract_series_seasons(api_response['seasons'], series_id),
```

Comment (elyse0) [reply yt_dlp/extractor/caracoltv.py]: That works for me n-n!

File: yt_dlp/extractor/caracoltv.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        app_token = self._extract_app_token(webpage)
+
+        bearer_token = self._download_json(
+            'https://eu-gateway.inmobly.com/applications/oauth', None, data=b'', note='Retrieving bearer token',
+            headers={'Authorization': f'Basic {app_token}'})['token']
+
+        self._USER_TOKEN = self._download_json(
+            'https://eu-gateway.inmobly.com/user/login', None, note='Performing login', headers={
+                'Content-Type': 'application/json',
+                'Authorization': 'Bearer' + bearer_token
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:76]:
```suggestion
                'Authorization': f'Bearer {bearer_token}',
```

File: yt_dlp/extractor/caracoltv.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            'season_number': int_or_none(season_number),
+            'episode_number': int_or_none(video_data.get('item_order')),
+            'is_live': video_data.get('entry_type') == 3,
+        }
+
+    def _extract_series_seasons(self, seasons, series_id):
+        for season in seasons:
+            api_response = self._download_json(
+                'https://eu-gateway.inmobly.com/feed', series_id, query={'season_id': season['id']},
+                headers={'Authorization': 'Bearer ' + self._USER_TOKEN})
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:112]:
```suggestion
                headers={'Authorization': f'Bearer {self._USER_TOKEN}'})
```

File: yt_dlp/extractor/caracoltv.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+    def _real_extract(self, url):
+        series_id = self._match_id(url)
+
+        if self._USER_TOKEN is None:
+            self._perform_login('guest@inmobly.com', 'Test@gus1')
+
+        api_response = self._download_json(
+            'https://eu-gateway.inmobly.com/feed', series_id, query={'include_ids': series_id},
+            headers={'Authorization': 'Bearer ' + self._USER_TOKEN})['items'][0]
```
Comment (bashonly) [inline yt_dlp/extractor/caracoltv.py:126]:
```suggestion
            headers={'Authorization': f'Bearer {self._USER_TOKEN}'})['items'][0]
```

File: yt_dlp/extractor/canal1.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        'url': 'https://canal1.com.co/programas/guerreros-canal-1/video-inedito-guerreros-despedida-kewin-zarate/',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        return self.url_result(
+            self._search_regex(r'"embedUrl"\s*:\s*"([^"]+)', webpage, "embed url"),
```
Comment (bashonly) [inline yt_dlp/extractor/canal1.py:38]:
```suggestion
            self._search_regex(r'"embedUrl"\s*:\s*"([^"]+)', webpage, 'embed url'),
```

**Review by bashonly — APPROVED** [review]

**Review by coletdjnz — APPROVED** [review]

**Pull request merged**